x-pack/plugins/apm/public/components/routing/templates/apm_service_template.tsx

```diff
@@ -100,6 +114,42 @@ function Template({
   );
 }
 
+function AnalyzeDataButton({ serviceName }: { serviceName: string }) {
+  const { agentName } = useApmServiceContext();
+  const { services } = useKibana();
+  const { urlParams } = useUrlParams();
+  const { rangeTo, rangeFrom, environment } = urlParams;
+  const basepath = services.http?.basePath.get();
+
+  if (isRumAgentName(agentName) || isIosAgent(agentName)) {
+    const href = createExploratoryViewUrl(
+      {
+        'apm-series': {
```

Comment:
Is this (apm-series) correct? What does it mean?

Reply:
it's just a name I guess

Reply:
yeah it's just a unique value. Not of consideration atm.

Reply:
Okay, then what's the reason for not using an array?

```
const href = createExploratoryViewUrl(
  [
    {
      dataType: isRumAgentName(agentName) ? "ux" : "mobile",
      time: { from: rangeFrom, to: rangeTo },
      reportType: "kpi-over-time",
      reportDefinitions: {
        [SERVICE_NAME]: [serviceName],
        ...(!!environment && ENVIRONMENT_NOT_DEFINED.value !== environment
          ? { [SERVICE_ENVIRONMENT]: [environment] }
          : {}),
      },
      operationType: "average",
      isNew: true,
    } as SeriesUrl,
  ],
  basepath
);

```

Reply:
Array totally makes sense, i will refactor it in a follow up.